### ISSUE TITLE
local - when using branch is producing a main-less repo. 

### ISSUE THREAD

**Issue by 6za** [kubefirst/kubefirst#818]
# Overview

Local, when build from branch is creating a mainless repo:
![image](https://user-images.githubusercontent.com/53096417/205368225-95469f3f-4dc6-468b-ba22-6dc83b2e3720.png)

Registry should be based on `main` to match cluster version.

**Comment by 6za** [comment]
Created using: 
```bash 
go build -ldflags="-X github.com/kubefirst/kubefirst/configs.K1Version=1.10.8" .
./kubefirst local --gitops-branch feat_813 --skip-metaphor
```

**Comment by 6za** [comment]
Reviewing the new local clone/checkout logic.. it seems to be missing this logic to add main from branches/tags: https://github.com/kubefirst/kubefirst/blob/1_11_release/internal/gitClient/git.go#L273

Local seems to be producing main-less repos at github.

**Comment by 6za** [comment]
It seems to missing some logic here, to address it:
- https://github.com/kubefirst/kubefirst/blob/1_11_release/internal/gitClient/git.go#L528
- https://github.com/kubefirst/kubefirst/blob/1_11_release/internal/gitClient/git.go#L510

I am not sure if this is by design, but it seems to be odd, to be `main-less` the repo. I believe the 1st branch will be made "HEAD" on this scenarios.

**Comment by 6za** [comment]
Merged to Release 1.11 branch
